image/cli/mascli/functions/gitops_deprovision_cluster

```diff
@@ -201,6 +201,16 @@ function gitops_deprovision_cluster() {
   export SECRET_NAME_DRO=${ACCOUNT_ID}/${CLUSTER_ID}/dro
   echo "Delete $SECRET_NAME_DRO from Secrets Manager"
   sm_delete_secret $SECRET_NAME_DRO
+  
+  # Created for custom service accounts
+  CLUSTER_SECRETS=($(sm_list_cluster_secrets ${ACCOUNT_ID} ${CLUSTER_ID}))
+  for SECRET_NAME in "${CLUSTER_SECRETS[@]}"; do
+    if [[ "$SECRET_NAME" == *"/custom-sa/"* ]]; then
```

Comment:
This deletes any cluster-level secret containing the string "/custom-sa/" right? Could we make this check a bit more specific to mitigate the risk of unintentionally deleting secrets if we ever use "custom-sa" as a path in a different cluster-level secret in future? 

Perhaps could do a regex check here, something like:
`^[^\/]+\/[^\/]+\/custom-sa\/[^\/]+$`? 

The `~=` bash operator might work with that regex, but you'll need to experiment.

Note: I've used `[^\/]+` instead of the actual account/cluster id strings (which are already checked due to the use of `sm_list_cluster_secrets`) so we don't have to worry about escaping any reserved regex characters that might be used in those IDs.

Reply:
Yes, this deletes any cluster level secret that contain "/custom-sa/" in the name. I agree with your approach to mitigate the risk of deleting the wrong secrets in the future. Regex check added!

Reply:
I tested changes to the `gitops_deprovision_cluster` function with the script below. I pulled the `sm_list_cluster_secrets` and `sm_delete_secret` functions from the cli code and added the code block for removing custom-sa secrets. 

**Steps for Testing**
1. Create test secrets in DEV AWS Secrets Manager.  
- `fyre-dev-noble7/noble7/custom-sa/massaas-ssp-sa`: should be deleted
- `fyre-dev-noble7/noble7/custom-sa/massaas-icd-sa`: should be deleted
- `fyre-dev-noble7/noble7/custom-sa/test-string/massaas-ssp-sa`: should not be deleted
- `fyre-dev-noble7/noble7/custom-sa`: should not be deleted
2. Execute test script `cluster_deprovision_test.sh`.
3. Validate that the correct secrets have been deleted from AWS Secrets Manager.

![Screenshot 2024-12-19 at 10 56 03 AM](https://github.com/user-attachments/assets/99135902-888c-4745-ad61-62b09f7b5a47)


**Test script `cluster_deprovision_test.sh`** 
```bash
#!/bin/bash
DELIMIITER="/"
AVP_TYPE="aws"
ACCOUNT_ID="fyre-dev-noble7"
CLUSTER_ID="noble7"


function sm_list_cluster_secrets() {
  ACCOUNT=$1
  CLUSTER_ID=$2

  if [[ "$AVP_TYPE" == "aws" ]]; then
    set +o pipefail
    LIST_SECRETS_OUTPUT="$(aws secretsmanager list-secrets --output yaml --no-cli-pager --filters Key=name,Values=${ACCOUNT}${DELIMIITER}${CLUSTER_ID}${DELIMIITER})"
    return_code=$?
    if [ $return_code -ne 0 ]; then
      echo "aws: secretmanager list-secrets failed with error: $return_code"
      exit $return_code
    fi
    set -o pipefail

    echo "$LIST_SECRETS_OUTPUT" | yq -r '.SecretList[].Name'
  elif [[ "$AVP_TYPE" == "ibm" ]]; then
    echo "IBM SecretsManager not yet supported"
    exit 1
  fi
}

function sm_delete_secret(){
  SECRET_NAME=$1
  if [[ "$AVP_TYPE" == "aws" ]]; then

    # Delete the secret
    set +o pipefail
    aws secretsmanager delete-secret --force-delete-without-recovery --secret-id ${SECRET_NAME} --output json 2> /dev/null || exit 1
    set -o pipefail

  elif [[ "$AVP_TYPE" == "ibm" ]]; then
    echo "IBM SecretsManager not yet supported"
    exit 1
  fi
}

CLUSTER_SECRETS=($(sm_list_cluster_secrets ${ACCOUNT_ID} ${CLUSTER_ID}))
PATTERN='^[^\/]+\/[^\/]+\/custom-sa\/[^\/]+$'
for SECRET_NAME in "${CLUSTER_SECRETS[@]}"; do
   if [[ "$SECRET_NAME" =~ $PATTERN ]]; then
    export SECRET_NAME_CUSTOM_SA=${SECRET_NAME}
    echo "Delete $SECRET_NAME_CUSTOM_SA from Secrets Manager"
    sm_delete_secret ${SECRET_NAME_CUSTOM_SA}
  fi
done
```

Reply:
Great work, thank you for the detailed overview. Approved.